fedimint-api/src/config.rs

```diff
@@ -52,6 +57,42 @@ pub struct ClientConfig {
     pub modules: BTreeMap<String, ClientModuleConfig>,
 }
 
+/// The federation id is a copy of the authentication threshold public key of the federation
+///
+/// Stable id so long as guardians membership does not change
+/// Unique id so long as guardians do not all collude
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, Hash, PartialEq)]
+pub struct FederationId(pub threshold_crypto::PublicKey);
+
+/// Display as a hex encoding
+impl FederationId {
+    /// Non-unique dummy id for testing
+    pub fn dummy() -> Self {
+        Self(threshold_crypto::PublicKey::from(G1Projective::identity()))
+    }
+
+    fn from_bytes(bytes: [u8; 48]) -> Self {
+        Self(threshold_crypto::PublicKey::from_bytes(bytes).expect("Invalid pub-key"))
```

Comment:
This `expect` is a potential DoS vector if the function is ever used to parse untrusted data, which might happen through the publicly accessible `FromStr` impl.

Reply:
@jkitman Yeah, we should `Result` it. :)

Reply:
https://github.com/fedimint/fedimint/pull/1222